app/static/src/app/store/graphs/actions.ts

```diff
@@ -5,21 +5,33 @@ import { FitDataAction } from "../fitData/actions";
 import { GraphsState } from "./state";
 
 export enum GraphsAction {
-    UpdateSelectedVariables = "UpdateSelectedVariables"
+    UpdateSelectedVariables = "UpdateSelectedVariables",
+    NewGraph = "NewGraph"
 }
 
 export const actions: ActionTree<GraphsState, AppState> = {
     UpdateSelectedVariables(context, payload: { graphIndex: number; selectedVariables: string[] }) {
         const { commit, dispatch, rootState } = context;
         // Maintain unselected variables too, so we know which variables had been explicitly unselected when model
         // updates
+        const allVariables = rootState.model.odinModelResponse?.metadata?.variables || [];
         const unselectedVariables =
             rootState.model.odinModelResponse?.metadata?.variables.filter(
                 (s) => !payload.selectedVariables.includes(s)
             ) || [];
-        commit(GraphsMutation.SetSelectedVariables, { ...payload, unselectedVariables });
+        // sort the selected variables to match the order in the model
+        const selectedVariables = payload.selectedVariables.sort((a, b) =>
```

Comment:
is there a benfit of the sort? is there significance in order of variables

Reply:
Yeah, this is the variable order that will be displayed in the UI, and we want it to match the order of variables as they're defined in the model, which might typically follow a temporal order e.g. S(usceptible), I(nfected), R(ecovered) or be ordered by category. Just leaving them unordered or sorting alphabetically would be less meanginful for the user. This matches the previous toggle behaviour where variables were always in model order.